airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -50,66 +53,107 @@ def build_include(self) -> List[str]:
             for dependency_directory in self.context.connector.get_local_dependency_paths(with_test_dependencies=True)
         ]
 
-    def _get_gradle_command(self, extra_options: Tuple[str, ...] = ("--no-daemon", "--scan", "--build-cache")) -> List:
-        command = (
-            ["./gradlew"]
-            + list(extra_options)
-            + [f":airbyte-integrations:connectors:{self.context.connector.technical_name}:{self.gradle_task_name}"]
-            + list(self.gradle_task_options)
+    def _get_gradle_command(self, task: str) -> List[str]:
+        return sh_dash_c(
+            [
+                # The gradle command is chained in between a couple of rsyncs which load from- and store to the cache volume.
+                "(rsync -a --stats /root/gradle-cache/ /root/.gradle || true)",
+                f"(./gradlew {' '.join(self.DEFAULT_GRADLE_TASK_OPTIONS)} {task} > >(tee /root/stdout) 2> >(tee /root/stderr >&2))",
+                "(rsync -a --stats /root/.gradle/ /root/gradle-cache || true)",
+            ]
         )
-        for task in self.DEFAULT_TASKS_TO_EXCLUDE:
-            command += ["-x", task]
-        return command
 
     async def _run(self) -> StepResult:
-        includes = self.build_include
-        if self.with_java_cdk_snapshot:
-            includes + ["./airbyte-cdk/java/airbyte-cdk/**"]
 
-        connector_under_test = (
-            environments.with_gradle(self.context, includes, bind_to_docker_host=self.BIND_TO_DOCKER_HOST)
+        include = [
+            ".root",
+            ".env",
+            "build.gradle",
+            "deps.toml",
+            "gradle.properties",
+            "gradle",
+            "gradlew",
+            "LICENSE_SHORT",
+            "settings.gradle",
+            "build.gradle",
+            "tools/gradle",
+            "spotbugs-exclude-filter-file.xml",
+            "buildSrc",
+            "tools/bin/build_image.sh",
+            "tools/lib/lib.sh",
+            "tools/gradle/codestyle",
+            "pyproject.toml",
+            "airbyte-cdk/java/airbyte-cdk/**",
+        ] + self.build_include
+
+        yum_packages_to_install = [
+            "docker",  # required by :integrationTestJava.
+            "findutils",  # gradle requires xargs, which is shipped in findutils.
+            "jq",  # required by :airbyte-connector-test-harnesses:acceptance-test-harness to inspect docker images.
+            "npm",  # required by :format.
+            "pip",  # required by :format.
```

Comment:
I bet you don't like that `format` requires non java packet managers like `npm` and `pip`

Reply:
I don't mind, tbh.